core/src/main/java/org/apache/iceberg/rest/RESTCatalog.java

```diff
@@ -55,7 +59,50 @@ public class RESTCatalog
   public RESTCatalog() {
     this(
         SessionCatalog.SessionContext.createEmpty(),
-        config -> HTTPClient.builder(config).uri(config.get(CatalogProperties.URI)).build());
+        config -> {
+          HTTPClient.Builder builder =
+              HTTPClient.builder(config).uri(config.get(CatalogProperties.URI));
+
+          String proxyHostname =
+              PropertyUtil.propertyAsString(
+                  config,
+                  CatalogProperties.PROXY_HOSTNAME,
+                  CatalogProperties.PROXY_HOSTNAME_DEFAULT);
```

Comment:
we can use `null` as the default and then check whether `proxyHostname` is null or not to determine whether the proxy hostname was set. Same for the other properties